test/Microsoft.AspNet.Mvc.Core.Test/ControllerActionInvokerTest.cs

```diff
@@ -52,6 +52,49 @@ private struct SampleStruct
         }
 
         [Fact]
+        public async Task InvokeAction_SavesTempData_WhenActionDoesNotThrow()
+        {
+            // Arrange
+            var filter = new Mock<IExceptionFilter>(MockBehavior.Strict);
+            filter
+                .Setup(f => f.OnException(It.IsAny<ExceptionContext>()));
+            var tempData = new Mock<ITempDataDictionary>();
+            tempData.Setup(t => t.Save()).Verifiable();
+
+            var invoker = CreateInvoker(filter.Object, actionThrows: false, tempData: tempData.Object);
+
+            // Act
+            await invoker.InvokeAsync();
+
+            // Assert
+            tempData.Verify(t => t.Save(), Times.Once());
+        }
+
+        [Fact]
+        public async Task InvokeAction_SavesTempData_WhenActionThrows()
+        {
+            // Arrange
+            var filter = new Mock<IExceptionFilter>(MockBehavior.Strict);
+            filter
+                .Setup(f => f.OnException(It.IsAny<ExceptionContext>()))
+                .Callback<ExceptionContext>(context =>
+                {
+                    // Handle the exception
+                    context.Result = new EmptyResult();
```

Comment:
I'd expect to see this just go unhandled for this test. Put an `Assert.Throws` around the invoker and don't add a filter